app/celery/celery.py

```diff
@@ -58,6 +58,8 @@ def apply_async(self, args=None, kwargs=None, task_id=None, producer=None,
 
             if has_request_context() and hasattr(request, 'request_id'):
                 kwargs['request_id'] = request.request_id
+            elif g.request_id:
```

Comment:
Oh, will this raise an attribute error if there is g does not have `request_id` as an attribute?

I just opened a `flask shell` and did `g.thing` and got an attribute error so wondering if we are at risk of the same here?